process_manager/views/partials.py

```diff
@@ -97,9 +97,13 @@ def process_table(request: HttpRequest) -> HttpResponse:
 def messages(request: HttpRequest) -> HttpResponse:
     """Renders Kafka messages from the database."""
     messages = []
-    for msg in DruncMessage.objects.all():
+
+    # Filter messages based on search parameter.
+    search = request.GET.get("search", "")
+    for msg in DruncMessage.objects.filter(message__icontains=search):
         # Time is stored as UTC. localtime(t) converts this to our configured timezone.
         timestamp = localtime(msg.timestamp).strftime("%Y-%m-%d %H:%M:%S")
```

Comment:
Not related to this PR, but `localtime` returns the actual local time _if such a time is set somewhere explicitl_y. Otherwise, Django will just return the default timezone - typically UTC. [Django does not autodetect the user's timezone](https://docs.djangoproject.com/en/5.1/topics/i18n/timezones/#selecting-the-current-time-zone), as far as I learnt lately.

Reply:
Oh strange. I thought it picked up the one defined in `drunc_ui/settings/settings.py`?

Reply:
As in [here](https://docs.djangoproject.com/en/5.1/topics/i18n/timezones/#time-zone-aware-output-in-templates), also `USE_TZ` and `TIME_ZONE` [here](https://docs.djangoproject.com/en/5.1/topics/i18n/timezones/#default-time-zone-and-current-time-zone).

Reply:
Precisely, and in there is says:

> You should set the current time zone to the end user’s actual time zone with [activate()](https://docs.djangoproject.com/en/5.1/ref/utils/#django.utils.timezone.activate). Otherwise, the default time zone is used.

But Django cannot get the user's actual timezone automatically, as explained exacly in the next section, that I linked above.